src/Math-Matrix/PMTensor.class.st

```diff
@@ -0,0 +1,152 @@
+Class {
+	#name : #PMTensor,
+	#superclass : #Object,
+	#instVars : [
+		'array',
+		'shape',
+		'first',
+		'strides'
+	],
+	#category : #'Math-Matrix'
+}
+
+{ #category : #'instance creation' }
+PMTensor class >> fromNestedArray: anArray [
+
+	^ self new array: anArray flattened withShape: (self shape:anArray)
+]
+
+{ #category : #'instance creation' }
+PMTensor class >> newWith: anInteger [
```

Comment:
How about `fromScalar:` for this method, in analogy to the newly named `fromNestedArray:`?